internal/operands/metrics/reconcile.go

```diff
@@ -96,7 +97,12 @@ func reconcileMonitoringRbacRoleBinding(request *common.Request) (common.Reconci
 }
 
 func reconcilePrometheusRule(request *common.Request) (common.ReconcileResult, error) {
-	prometheusRule, err := newPrometheusRule(request.Namespace)
+	err := rules.SetupRules()
```

Comment:
Sure just where is the start of the operator?

Reply:
You can move it to the place were prometheus server is started:
https://github.com/kubevirt/ssp-operator/blob/1d127307da958cf7f5e345d028e77a167fe13983/main.go#L184

Or to the setup function that creates controllers:
https://github.com/kubevirt/ssp-operator/blob/cbca066af257bf181146edd2e9d7342ed6e0e578/controllers/setup.go#L50

Reply:
It's failing the tests since the rules is not available..

Reply:
Strange. I will debug it to find out why.

Reply:
I see. Unit tests are failing, because the rules are not registered during their execution.
Can you modify the `BeforeEach()` so some rules are registered?

By the way, I don't like that the library `github.com/machadovilaca/operator-observability` uses global variables for metrics. It makes it harder to write unit tests.

Reply:
fixed thank you so much!